pytorch_lightning/core/decorators.py

```diff
@@ -11,12 +11,18 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Decorator for LightningModule methods."""
+from pytorch_lightning.utilities import rank_zero_deprecation
 
-from functools import wraps
-from typing import Callable
+rank_zero_deprecation(
+    "Using ``pytorch_lightning.core.decorators.parameter_validation`` is deprecated in v1.5, "
+    "and will be removed in v1.7. It has been replaced by automatic parameters tying with "
+    "``pytorch_lightning.utilities.params_tying.set_shared_parameters``"
```

Comment:
I know it's confusing, but there is a rule:
- in code comments, warning, error and info messages we use markdown notation, so single backtick
- in sphinx, we use rst formatting -> double backticks

Reply:
Also note, it's pointing to the wrong import, should be `parameter_typing` not `params_tying`

Reply:
- should we mark parameter_validation as deprecated too? is it something that should also sit in the TPU plugins? 
- if parameter validation is deprecated, should we mark `core/decorators.py` as deprecated too? i'm not sure where else this is imported

Reply:
`parameter_validation` has been deprecated in this PR as well